src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,121 @@
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPostId,
+}) => {
+  const [details, setDetails] = useState<Post | null>(null);
+  const [comments, setComments] = useState<PostComment[]>([]);
+  const [isCommentsVisible, setIsCommentVisible] = useState(true);
+  const [commentLoading, setCommentLoading] = useState(false);
+
+  const getDetailsFromServer = useCallback(async () => {
+    const [detailsFromServer, commentsFromServer] = await Promise.all([
+      getPostDetails(selectedPostId),
+      getPostComments(selectedPostId),
+    ]);
+
+    await setDetails(detailsFromServer);
+    await setComments(commentsFromServer);
+    setCommentLoading(false);
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    setDetails(null);
+    setComments([]);
+    getDetailsFromServer();
+  }, [selectedPostId]);
+
+  const handleCommentVisible = useCallback(() => {
+    setIsCommentVisible(prev => !prev);
+  }, []);
+
+  const handleAddComment = useCallback(async (newComment: NewComment) => {
+    setCommentLoading(true);
+    const createdComment = await createComment(newComment);
+    const newComments = await [...comments, createdComment];
+
+    setComments(newComments);
+    setCommentLoading(false);
+  }, [comments]);
+
+  const handleDeleteComment = useCallback(async (deleteId: number) => {
+    setCommentLoading(true);
+    await deletePostComment(deleteId);
+    const filteredComments = [...comments].filter(
+      comment => comment.id !== deleteId,
+    );
+
+    setComments(filteredComments);
+    setCommentLoading(false);
+  }, [comments]);
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      {details ? (
+        <>
+          <section className="PostDetails__post">
+            <p>{details.body}</p>
+          </section>
+
+          {(comments.length > 0) && (
```

Comment:
on delete and add comment, comments not empty 

```suggestion
          {(comments.length > 0 && !commentLoading) && (
```